db/compaction_job.cc

```diff
@@ -901,27 +901,32 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
   RecordDroppedKeys(c_iter_stats, &sub_compact->compaction_job_stats);
   RecordCompactionIOStats();
 
-  if (status.ok() &&
-      (shutting_down_->load(std::memory_order_acquire) || cfd->IsDropped())) {
+  if (status.ok() && (shutting_down_->load() || cfd->IsDropped())) {
```

Comment:
It's intentional. The idea was to have a memory barrier here; otherwise, maybe, CompactionJob can finish with ShuttingDown status while some other thread still sees shutting_sown_=false. It's unlikely to be possible (there are certainly some memory barriers on the path from ProcessKeyValueCompaction to returning from CompactionJob::Run()), this way just seemed easier to reason about. Apparently it adds confusion instead, so I'll add the memory_order_relaxed.